mqtt_subscriber/device_profile_player/main.py

```diff
@@ -0,0 +1,65 @@
+import os
+import logging
+import json
+import asyncio
+from time import time
+import websockets
+from pendulum import now, duration
+from d3a_interface.utils import from_timestamp
+from d3a_interface.read_user_profile import read_profile_without_config
+
+
+class DeviceProfilePlayback:
+
+    def __init__(self):
+        profile_path = os.environ['DEVICE_PLAYBACK_PROFILES_PATH']
+        self._year_to_replay = int(os.environ['DEVICE_PLAYBACK_YEAR_TO_REPLAY'])
+        csv_files = self._list_all_available_filenames_and_paths(profile_path)
+        self.profile_dicts = {
+            filename.split(".")[0]: read_profile_without_config(filepath)
+            for filename, filepath in csv_files
+        }
+
+    @staticmethod
+    def _list_all_available_filenames_and_paths(profile_path):
+        filenames = os.listdir(profile_path)
+        return [(filename, os.path.join(profile_path, filename))
+                for filename in filenames
+                if filename.endswith(".csv")]
+
+    async def _scheduled_call(self, websocket):
+        current_timestamp = now().timestamp()
+        offset_seconds = current_timestamp % 900
+        requested_datetime = from_timestamp(current_timestamp - offset_seconds)\
+            .replace(year=self._year_to_replay)
```

Comment:
This is really just a precaution, right? `sleep_until` should deal with waiting a minute to resume, right?

Reply:
Actually this is something the simulator needs. At the moment, the profiles that we use contain values for past years (2020). The simulator will need to run in realtime though. Therefore, we need a way to detect which year from the profiles do we want to simulate. I know that this can be easily solved not in code, but by preprocessing the files and changing the dates to 2021, however the change was easy enough to incorporate into the code.

Reply:
Actually I mean the `from_timestamp(current_timestamp - offset_seconds)`. Not the year_to_replay.

Reply:
Well, this is also required too :D The simulator is supposed to have a 15-min cycle, but it is not the same as the CN, because it does not run at exactly 00:15, 00:30. 00:45 .... but at any time in between. This mimics the behavior of external measurement sources, since from what we have seen they are not 100% consistent in time. 
Therefore, assuming that the simulator cycle is on 00:20, 00:35, 00:50 ... there are no values at the profiles for these timestamps, because we quantize them to exact 15 min intervals. So I have to use the previous time slot, which makes it obligatory to map 00:20 -> 00:15, 00:35->00:30 etc.

Reply:
I am stupid. I thought that 900 seconds are one minute (OMG!) Now I fully understand and of course agree.
Please make these 900 seconds also configurable in the frame of https://github.com/gridsingularity/d3a-api-client/pull/117/files/bb36be7be7d4c05e1c8aeb9180367818e7beefc6#r595065235 Thanks a lot!

Reply:
You are not, it is essentially a change that hides much more than it seems from a first glance. Already made this configurable via CLI, thanks!